be/src/olap/task/build_inverted_index.cpp

```diff
@@ -0,0 +1,461 @@
+}
+
+BuildInvertedIndex::~BuildInvertedIndex() {
+    _olap_data_convertor.reset();
+    _inverted_index_builders.clear();
+}
+
+Status BuildInvertedIndex::init() {
+    _cur_tablet_schema = std::make_shared<TabletSchema>();
+    _cur_tablet_schema->update_tablet_columns(*_tablet->tablet_schema(), _columns);
+    for (auto inverted_index : _alter_inverted_indexes) {
+        auto column_name = inverted_index.columns[0];
+        auto column_idx = _cur_tablet_schema->field_index(column_name);
+        if (column_idx < 0) {
+            LOG(WARNING) << "referenced column was missing. "
+                         << "[column=" << column_name << " referenced_column=" << column_idx
+                         << "]";
+            return Status::Error<ErrorCode::CE_CMD_PARAMS_ERROR>();
+        }
+        auto column = _cur_tablet_schema->column(column_idx);
+        _alter_column_uids.emplace_back(column.unique_id());
+    }
+    return Status::OK();
+}
+
+Status BuildInvertedIndex::update_inverted_index_info() {
+    // just do link files
+    LOG(INFO) << "begin to update_inverted_index_info, tablet=" << _tablet->tablet_id()
+            << ", is_drop_op=" << _is_drop_op;
+    for (auto i = 0; i < _input_rowsets.size(); ++i) {
+        auto input_rowset = _input_rowsets[i];
+        TabletSchemaSPtr output_rs_tablet_schema = std::make_shared<TabletSchema>();
+        output_rs_tablet_schema->copy_from(*input_rowset->tablet_schema());
+        if (_is_drop_op) {
+            output_rs_tablet_schema->update_indexes_from_thrift(_exist_indexes);
+        } else {
+            for (auto t_inverted_index : _alter_inverted_indexes) {
+                TabletIndex index;
+                index.init_from_thrift(t_inverted_index, *_cur_tablet_schema);
+                output_rs_tablet_schema->append_index(std::move(index));
+            }
+        }
+        // construct input rowset reader
+        RowsetReaderSharedPtr input_rs_reader;
+        RETURN_IF_ERROR(input_rowset->create_reader(&input_rs_reader));
+        
+        // construct output rowset writer
+        std::unique_ptr<RowsetWriter> output_rs_writer;
+        RowsetWriterContext context;
+        context.version = input_rs_reader->version();
+        context.rowset_state = VISIBLE;
+        context.segments_overlap = input_rs_reader->rowset()->rowset_meta()->segments_overlap();
+        context.tablet_schema = output_rs_tablet_schema;
+        context.newest_write_timestamp = input_rs_reader->newest_write_timestamp();
+        context.fs = input_rs_reader->rowset()->rowset_meta()->fs();
+        Status status = _tablet->create_rowset_writer(context, &output_rs_writer);
+        if (!status.ok()) {
+            return Status::Error<ErrorCode::ROWSET_BUILDER_INIT>();
+        }
+        std::set<int32_t> alter_column_ids_set(_alter_column_uids.begin(), _alter_column_uids.end());
+        RETURN_IF_ERROR(input_rowset->link_files_without_index_to(
```

Comment:
Where do we check existence of index if _alter_column_uids are not linked?

Reply:
changed this logic